src/mappings/SchemeFactory/mapping.ts

```diff
@@ -0,0 +1,101 @@
+import { Address, BigInt, Bytes } from '@graphprotocol/graph-ts';
+
+// Import event types from the Reputation contract ABI
+import {
+  NewSchemeProposal,
+  ProposalExecuted,
+} from '../../types/SchemeFactory/SchemeFactory';
+
+import * as domain from '../../domain';
+
+// Import entity types generated from the GraphQL schema
+import {
+   SchemeFactoryNewSchemeProposal,
+   SchemeFactoryProposal,
+   SchemeFactoryProposalExecuted,
+} from '../../types/schema';
+import { equalsBytes, equalStrings, eventId } from '../../utils';
+
+export function handleNewSchemeProposal(event: NewSchemeProposal): void {
+  let ent = SchemeFactoryNewSchemeProposal.load(eventId(event));
+  if (ent == null) {
+    ent = new SchemeFactoryNewSchemeProposal(eventId(event));
+  }
+  ent.txHash = event.transaction.hash;
+  ent.contract = event.address;
+  ent.avatar = event.params._avatar;
+  ent.proposalId = event.params._proposalId;
+  ent.schemeName = event.params._schemeName;
+  ent.schemeData = event.params._schemeData;
+  ent.packageVersion = event.params._packageVersion;
+  ent.permission = event.params._permissions;
+  ent.schemeToReplace = event.params._schemeToReplace;
+  ent.descriptionHash = event.params._descriptionHash;
+  ent.votingMachine = event.params._intVoteInterface;
+  ent.save();
+
+  domain.handleNewSchemeFactoryProposal(event.params._proposalId.toHex(),
+                                         event.block.timestamp,
+                                         ent.avatar,
+                                         ent.votingMachine,
+                                         ent.descriptionHash,
+                                         event.address);
+  insertNewProposal(ent.avatar as Address,
+                          ent.proposalId,
+                          ent.schemeName,
+                          ent.schemeData,
+                          ent.packageVersion as BigInt[],
+                          ent.permission,
+                          ent.schemeToReplace as Address);
+}
+
+export function handleProposalExecuted(event: ProposalExecuted): void {
+  let ent = SchemeFactoryProposalExecuted.load(eventId(event));
+  if (ent == null) {
```

Comment:
is it not always null ?

Reply:
I'm pretty sure you have to have these checks even when they seem futile, if I remember correctly that's required due to some Graph Node special requirements.

Reply:
if that is the case so it is an issue with the graph , which give same event id for multiple events.
using porposalId is better..